packages/math-input/src/components/keypad/mobile-keypad.tsx

```diff
@@ -0,0 +1,158 @@
+
+    state: State = {
+        active: false,
+    };
+
+    activate: () => void = () => {
+        this.setState({active: true});
+    };
+
+    dismiss: () => void = () => {
+        this.setState({active: false});
+    };
+
+    configure: (configuration: KeypadConfiguration, cb: () => void) => void = (
+        configuration,
+        cb,
+    ) => {
+        this.setState({keypadConfig: configuration});
+
+        // HACK(charlie): In Perseus, triggering a focus causes the keypad to
+        // animate into view and re-configure. We'd like to provide the option
+        // to re-render the re-configured keypad before animating it into view,
+        // to avoid jank in the animation. As such, we support passing a
+        // callback into `configureKeypad`. However, implementing this properly
+        // would require middleware, etc., so we just hack it on with
+        // `setTimeout` for now.
+        setTimeout(() => cb && cb());
+    };
+
+    setCursor: (cursor: Cursor) => void = (cursor) => {
+        this.setState({cursor});
+    };
+
+    setKeyHandler: (keyHandler: KeyHandler) => void = (keyHandler) => {
+        this.setState({keyHandler});
+    };
+
+    getDOMNode: () => ReturnType<typeof ReactDOM.findDOMNode> = () => {
+        return ReactDOM.findDOMNode(this);
+    };
+
+    _handleClickKey(key: Key) {
+        if (key === "DISMISS") {
+            this.dismiss();
+            return;
+        }
+
+        const cursor = this.state.keyHandler?.(key);
+        this.setState({cursor});
+    }
+
+    render(): React.ReactNode {
+        const {active, cursor, keypadConfig} = this.state;
+
+        const containerStyle = [
+            styles.keypadContainer,
+            active ? styles.activeKeypadContainer : null,
+        ];
+
+        const isExpression = keypadConfig?.keypadType === "EXPRESSION";
+
+        return (
+            <View
+                style={containerStyle}
+                ref={(element) => {
+                    if (!this.hasMounted && element) {
+                        // Append the dispatch methods that we want to expose
+                        // externally to the returned React element.
+                        const elementWithDispatchMethods = {
+                            ...element,
+                            activate: this.activate,
+                            dismiss: this.dismiss,
+                            configure: this.configure,
+                            setCursor: this.setCursor,
+                            setKeyHandler: this.setKeyHandler,
+                            getDOMNode: this.getDOMNode,
+                        } as const;
+
+                        this.hasMounted = true;
+                        this.props.onElementMounted?.(
+                            elementWithDispatchMethods,
+                        );
+                    }
+                }}
+            >
+                <Keypad
+                    sendEvent={async () => {}}
```

Comment:
Can you leave a `// TODO(jeremy)` here?  This analytics has taken waaaay too long. 😭